dist/src/main/java/io/camunda/zeebe/gateway/RestApiCompositeFilter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import org.springframework.http.MediaType;
+import org.springframework.web.filter.CompositeFilter;
+
+public class RestApiCompositeFilter extends CompositeFilter {
+
+  public RestApiCompositeFilter(final List<Filter> filters) {
+    setFilters(filters);
+  }
+
+  @Override
+  public void doFilter(
+      final ServletRequest request, final ServletResponse response, final FilterChain chain)
+      throws IOException, ServletException {
+    try {
+      super.doFilter(request, response, chain);
+    } catch (final Exception e) {
+      final HttpServletRequest servletRequest = (HttpServletRequest) request;
+      final HttpServletResponse servletResponse = (HttpServletResponse) response;
```

Comment:
❓ Should we ensure we're not running into ClassCastExceptions here so we don't mask any original exceptions accidentally? Not entirely sure what to do when those aren't `HttpServletRequest` and response, maybe leave the `instance` part of the `msg` empty? 🤔

Reply:
> Not entirely sure what to do when those aren't `HttpServletRequest` and response

If they aren't instances of `HttpServletRequest/HttpServletResponse` we wouldn't be able to provide:
* the HTTP status code
* the request URI

@tmetzke maybe we can return some limited response (if we can) about internal issues and log an error.

Reply:
Let's provide as much information as possible. We can construct the `msg` as is, without the request URI. The response can only write the `msg` and not receive a status but so be it. Wdyt?

Reply:
👍 Have a look at what I did with [d823a46](https://github.com/camunda/camunda/pull/19430/commits/d823a4680dffa41638739e8cc61c299ba4a2beba).

Reply:
We're treating the non-`HttpServletRequest` (and response) case like an exception now. Not sure if that is necessary. How about we use something like the following:
```java
      final var instance = request instanceof HttpServletRequest ? ((HttpServletRequest) request).getRequestURI() : "";
      final var msg =
          "{ \"type\": \"about:blank\", \"status\": 500, \"title\": \"Filter issue\", \"detail\": \""
              + e.getMessage()
              + "\", \"instance\": \" "
              + servletRequest.getRequestURI()
              + "\" }";

      if (response instanceof HttpServletResponse) {
        ((HttpServletResponse) response).setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
      }

      response.setContentType(MediaType.APPLICATION_PROBLEM_JSON_VALUE);
      response.getWriter().write(msg);
```

Reply:
My goal was to catch any occurrences of a non-HttpServletRequest (and response) case. But this will achieve almost the same without the extra code. I'll adjust it.

Reply:
@tmetzke all done 😄

Reply:
🚀